snap/hooks/connect-plug-graphics-core22

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash -e
+
+# Ensure nvidia support setup correctly #
+if snapctl is-connected nvidia-core22:graphics-core22 ; then
```

Comment:
it's not ok to mix bases like this. nvidia-core22 is only meant to be used strictly with snaps that are also base:core22.

Reply:
Hmm, ok.  I'm not sure how to work around that then.  We are using the docker snap on both UC20 and UC22.

Reply:
Would it be true to say then, for a docker snap with nvidia support it would have to be built for each required UC version base ?

I guess that could potentially still be solved by git branch mapped to snap channel.

Reply:
no, not quite. It probably is best to compile the require tool, as a part inside the docker snap, and vendor the require library it needs. Such that it is self contained, and can upgrade the tool and libraries to be compatible with what it has. And that way it should be self contained.

Reply:
also the nvidia-core22 libraries may not be compatible with the contents of the containers, I do expect containers to potentially contain their own userspace libraries. And I expected for host side to not need any of the sort to get things going. Will study the nvidia-ctk more to see what can be done there.

Reply:
@xnox I had a quick go at switching to a core22 base.  Not too many changes required, and it does seem to work.  I haven't included the changes in this PR yet, but for ref they are in a separate branch, called `core22-nvidia-support`

Here's the diff:
- https://github.com/jocado/docker-snap/compare/nvidia-support...jocado:docker-snap:core22-nvidia-support